### PR TITLE
clean up oncomatrix launcher that cgconly option is no longer generat…

### PR DESCRIPTION
**Pull request by xzhou82**
…ed by this code

## Description

i was about to make a launchGDCexp.js so I reviewed launchGdcMatrix.js and did some clean up. no UI change is introduced

## Checklist

[Check each task](https://github.com/stjude/proteinpaint/wiki/Pull-Request-Checklist) that has been performed or verified to be not applicable.
- [ ] Tests: added and/or passed unit and integration tests, or N/A
- [ ] Todos: commented or documented, or N/A
- [ ] Notable Changes: updated release.txt, prefixed a commit message with "fix:" or "feat:", added to an internal tracking document, or N/A

### CODE REVIEW

File: client/src/launchGdcMatrix.js [hunk 1]
```diff
@@ -219,16 +207,16 @@ async function getGenes(arg, gdcCohort, CGConly, maxGenes = 50) {
...
+		geneFilter: matrix.geneFilter
 	}
-	if (CGConly) body.CGConly = 1
 	const data = await dofetch3('gdc_filter2topGenes', { body })
 	if (data.error) throw data.error
 	if (!data.genes) throw 'no top genes found using the cohort filter'
 	return await Promise.all(
 		// do tempfix of "data.genes.slice(0,3).map" for faster testing
-		data.genes.map(async i => {
+		data.genes.slice(0, 3).map(async i => {
```
Comment (xzhou82) [inline client/src/launchGdcMatrix.js:219]: must delete before merging

File: client/src/launchGdcMatrix.js [hunk 2]
```diff
@@ -137,14 +148,11 @@ export async function init(arg, holder, genomes) {
 							chartType: 'matrix',
 							settingsKey: 'maxGenes',
 							callback: async value => {
```
Comment (xzhou82) [inline client/src/launchGdcMatrix.js:150]: i think this max number gene limit should not be defined here as a custom input. it should be coded in matrix and applied to other dataset. can assign to Congyu?

**Comment by xzhou82** [comment]
@airenzp need help to update `dom/genesetEdit.js` around line 50 on the optional CGC checkbox. please pull sjpp master to see the change (sorry accidentally committed to master, was to make a PR)

>                 topMutatedGenes: {
> -                       arguments: [{ id: 'CGConly', label: 'Use only cancer census genes', type: 'boolean', value: true }]
> +                       arguments: [{ id: 'geneFilter', label: 'Use only cancer census genes', type: 'string', value: 'CGC' }]

**Review by karishma-gangwani — APPROVED** [review]

**Review by karishma-gangwani — APPROVED** [review]
all tests pass.

**Comment by xzhou82** [comment]
CGC only option is no longer showing due to sjpp change, i would like Airen to fix it in this branch
also want to document args {settings, matrix} from Edgar

File: client/src/launchGdcMatrix.js [hunk 3]
```diff
@@ -1,31 +1,43 @@
...
+		// and is supplied by the OncoMatrixWrapper in the GFF repo, no need to supply here
+		// return { op: 'in', content: { field: 'cases.disease_type', value: ['Gliomas'] } }
 
 	.genes[]
 		future: list of genes to launch matrix
 
+	.termgroups[]
+		TODO 
+	
+	.settings{}
```
Comment (xzhou82) [inline client/src/launchGdcMatrix.js:36]: @siosonel could you comment on the reason for needing two attributes `arg.settings.matrix{}, arg.opts.matrix{}`

Comment (siosonel) [reply client/src/launchGdcMatrix.js:36]: @xzhou82 
1. `arg.settings.matrix`: any state that can be JSON-encoded/hydrated as part of `getPlotConfig().settings.matrix`. This is meant to hold plot-instance-specific state that only affects rendering and not data requests, but that distinction is not as clear with plots that use server-side rendering like violin plot.

2. `arg.plots.matrix`:  any options that cannot be JSON-stringified like callbacks. These options will also be applied to all rehydrated or new plots of the same type, so not plot-instance-specific.

File: client/src/launchGdcMatrix.js (hunk 2 above)
Comment (siosonel) [reply client/src/launchGdcMatrix.js:150]: @congyu-lu we can discuss how to move this code to matrix.

**Comment by airenzp** [comment]
Hi. I am looking at it Xin. I see the parameter type changed from a boolean to a string, so do you expect a string to be entered instead of a boolean?

**Comment by airenzp** [comment]
I pushed a fix to handle the parameter and added a PR in sjpp that should be merged with this one.

**Pull request merged**